app/jobs/create_transcoded_item_job.rb

```diff
@@ -0,0 +1,21 @@
+class CreateTranscodedItemJob < ApplicationJob
+  queue_as :within_30_seconds
+
+  def perform(audio_file, codec_conversion)
+    # Check that this transcoded_item was not created while this job was in the queue
+    return if TranscodedItem.find_by(audio_file:, codec_conversion:).present?
+
+    path = TranscodedItem.random_path
+    FileUtils.mkdir_p Pathname.new(path).parent
+    audio_file.convert(codec_conversion, path)
```

Comment:
Since we have a unique index on these paths, this could potentially:
* return a uuid that was already used for track A
* override the output of the existing transcode A with transcode B
* fail to save transcode B because of the unique index
* create a lot of confusion, since transcode A now returns audio for B

Maybe `TranscodedItem.random_path` should return a path that was not yet used?

Reply:
Good point.